compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1445,25 +1445,26 @@ impl<F: fmt::Write> Printer<'tcx> for FmtPrinter<'_, 'tcx, F> {
             return Ok(self);
         }
 
+        let name = match disambiguated_data.data.get_name() {
+            DefPathDataName::Named(name) => name,
+            DefPathDataName::Anon { namespace } => namespace,
+        };
```

Comment:
This is somewhat unnecessary, you can just rewrite the use below to check for `DefPathData::Named(kw::Invalid)`.